addons/bus/static/src/logger.js

```diff
@@ -0,0 +1,67 @@
+export class Logger {
+    static LOG_TTL = 45 * 60 * 1000; // 45 minutes
+    static instances = new WeakSet();
+
+    static purgeAllExpiredLogs() {
+        return Promise.all(Array.from(this.instances).map((l) => l.purgeExpiredLogs()));
+    }
+
+    constructor(name) {
+        this._name = name;
+        this._openDbPromise = new Promise((resolve) => {
+            const request = indexedDB.open(this._name, 1);
+            request.onsuccess = (event) => {
+                Logger.instances.add(this);
+                resolve(event.target.result);
+            };
+            request.onupgradeneeded = (event) => {
+                const db = event.target.result;
+                if (!db.objectStoreNames.contains("logs")) {
+                    db.createObjectStore("logs", { autoIncrement: true });
+                }
+            };
+        });
+    }
+
+    async log(message) {
+        const db = await this._openDbPromise;
+        const transaction = db.transaction("logs", "readwrite");
+        const store = transaction.objectStore("logs");
+        const addRequest = store.add({ timestamp: Date.now(), message });
+        return new Promise((res, rej) => {
+            addRequest.onsuccess = res;
+            addRequest.onerror = rej;
+        });
+    }
+
+    async getLogs() {
+        await Logger.purgeAllExpiredLogs();
+        const db = await this._openDbPromise;
+        const transaction = db.transaction("logs", "readwrite");
```

Comment:
I think write is not useful inside this function, so maybe `readonly`

Reply:
Good catch, it was readwrite for debugging purposes :)